.tree-sitter-lint/local_rules/src/lib.rs

```diff
@@ -0,0 +1,15 @@
+use std::sync::Arc;
+
+use tree_sitter_lint::Rule;
+
+mod rules;
+
+use rules::{no_default_default_rule, no_lazy_static_rule, prefer_impl_param_rule};
+
+pub fn get_rules() -> Vec<Arc<dyn Rule>> {
```

Comment:
This what I'm thinking the very basic "plugin API" looks like, expose a `Vec<Arc<dyn Rule>>` under a known "entry-point" name?

Reply:
Do they need to be in an `Arc`? (Could the linter just stick them in an `Arc` later itself? I guess they'd still have to be boxed, though.)

Reply:
I think it would be hard to make them have "single ownership" (and use borrow-checker semantics from there) eg in what I think is a later PR I make it so that you can get visibility of the underlying `Rule` from the `RuleInstance`/`RuleInstancePerFile` and so it seems simpler for them to have their own "owning" references ("up-front")?